spring-integration-core/src/main/java/org/springframework/integration/store/AbstractMessageGroupStore.java

```diff
@@ -56,11 +68,15 @@ public abstract class AbstractMessageGroupStore extends AbstractBatchingMessageG
 
 	private boolean timeoutOnIdle;
 
+	protected LockRegistry lockRegistry;
```

Comment:
This must be `private` and whenever we use outside of this class has to be replaced by the `protected getLockRegistry()`.

Reply:
You are right, even LockRegister does not need to be exposed to subclasses. But SimpleMessageStore is special, it needs lockRegister and has its own semaphore mechanism. I hope you can handle the follow-up work, because I feel there will be many special considerations in it. I will not update this PR again

Reply:
Please, confirm that you are OK that I'll pick it up whenever you have left.

I'll do that, thought, in new year, when I switch to `6.5` already.

Happy holidays!